src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -378,13 +395,30 @@ export class TabsTitleControl extends TitleControl {
 				if (duplicates.length > 1) {
 					const shortenedDescriptions = shorten(duplicates.map(duplicate => duplicate.description));
 					duplicates.forEach((duplicate, i) => {
-						duplicate.description = shortenedDescriptions[i];
-						duplicate.hasAmbiguousName = true;
+						mapLabelAndDescriptionToDifferential.set(
+							`${duplicate.name}${duplicate.description}`,
+							shortenedDescriptions[i],
+						);
 					});
 				}
 			}
 		});
 
+		for (const label of labels) {
```

Comment:
This seems to be a very specific way of treating description labels by using the native path separator. You have to keep in mind though that there are editors that are not about files (e.g. extension editor). 

I think a better approach is to push the logic of a short, medium or long description to each `EditorInput`.  If you see how it is done for the window title ([`FileEditorInput.getTitle()`](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/common/editors/fileEditorInput.ts#L145)) we should do something similar for the description imho. We can just change the `getDescription` signature to accept short, medium or long version (we already have a verbose flag which is going into that direction).

Reply:
That does sound better, but it's more involved than I can commit to.

Reply:
As I mentioned before, just move the logic into `FileEditorInput`, no need to implement it for all other editors.